src/parquet/column/reader.h

```diff
@@ -184,7 +188,7 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
     if (num_def_levels != num_rep_levels) {
       throw ParquetException("Number of decoded rep / def levels did not match");
```

Comment:
@edani in this case you need to know for sure that your data does not have any null values. An arbitrary Parquet reader can't assume this, of course, so I'm find leaving in the "short circuit" to save the computational work as long as it's well documented.